### PR TITLE
libtorrent: add version 2.0.8, support conan v2, update dependencies

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **libtorrent/***

- add version 2.0.8
- support conan v2
- update dependencies
- remove unused patches

notice: libtorrent 2.0.{0..6} can't use recent boost

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jwillikers — DISMISSED** [review]

File: recipes/libtorrent/all/conanfile.py [hunk 1]
```diff
@@ -45,126 +49,129 @@ class LibtorrentConan(ConanFile):
...
         else:
             self._check_compiler_supports_cxx14()
             if self.settings.compiler.get_safe("cppstd"):
-                tools.check_min_cppstd(self, 14)
+                check_min_cppstd(self, 14)
 
     def requirements(self):
-        if tools.Version(self.version) < "2.0.0":
-            self.requires("boost/1.79.0")
+        # libtorrent 2.0.x [x<=6] have issue for recent boost https://github.com/arvidn/libtorrent/discussions/6757
```
Comment (uilianries) [inline recipes/libtorrent/all/conanfile.py:87]: Thank you for detailing it!

**Review by uilianries — APPROVED** [review]
LGTM

File: recipes/libtorrent/all/conanfile.py [hunk 2]
```diff
@@ -45,126 +49,129 @@ class LibtorrentConan(ConanFile):
...
         if self.options.enable_encryption:
-            self.requires("openssl/1.1.1q")
+            self.requires("openssl/1.1.1s")
         if self.options.enable_iconv:
             self.requires("libiconv/1.17")
 
     def _validate_dependency_graph(self):
-        if tools.Version(self.deps_cpp_info["boost"].version) < "1.69.0" and \
+        if Version(self.dependencies["boost"].ref.version) < "1.69.0" and \
            (self.options["boost"].header_only or self.options["boost"].without_system):
```
Comment (SpaceIm) [inline recipes/libtorrent/all/conanfile.py:99]:
```suggestion
           (self.dependencies["boost"].options.header_only or self.dependencies["boost"].options.without_system):
```

File: recipes/libtorrent/all/conanfile.py [hunk 3]
```diff
@@ -45,126 +49,129 @@ class LibtorrentConan(ConanFile):
...
                                   replace)
             if self.settings.compiler == "clang" and self.settings.compiler.libcxx == "libstdc++":
                 # https://github.com/arvidn/libtorrent/issues/3557
-                tools.replace_in_file(os.path.join(self._source_subfolder, "include", "libtorrent", "file_storage.hpp"),
+                replace_in_file(self, os.path.join(self.source_folder, "include", "libtorrent", "file_storage.hpp"),
                                       "file_entry& operator=(file_entry&&) & noexcept = default;",
                                       "file_entry& operator=(file_entry&&) & = default;")
 
     def build(self):
         self._validate_dependency_graph()
```
Comment (SpaceIm) [inline recipes/libtorrent/all/conanfile.py:161]: to move to validate() now

File: recipes/libtorrent/all/conanfile.py [hunk 4]
```diff
@@ -1,10 +1,14 @@
...
-from conans import CMake, ConanFile, tools
-from conans.errors import ConanInvalidConfiguration
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.microsoft import check_min_vs, is_msvc_static_runtime, is_msvc, msvc_runtime_flag
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy, rm, rmdir, replace_in_file
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.env import VirtualBuildEnv
```
Comment (SpaceIm) [inline recipes/libtorrent/all/conanfile.py:8]: many unused imports according to linter

File: recipes/libtorrent/all/conanfile.py [hunk 5]
```diff
@@ -45,126 +49,129 @@ class LibtorrentConan(ConanFile):
...
-        self._cmake.definitions["python-bindings"] = False
-        if self._is_msvc:
-            self._cmake.definitions["static_runtime"] = "MT" in msvc_runtime_flag(self)
-        self._cmake.configure()
-        return self._cmake
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["Boost_USE_STATIC_LIBS"] = not self.options["boost"].shared
```
Comment (SpaceIm) [inline recipes/libtorrent/all/conanfile.py:122]:
```suggestion
        tc.variables["Boost_USE_STATIC_LIBS"] = not self.dependencies["boost"].options.shared
```

File: recipes/libtorrent/all/conanfile.py [hunk 6]
```diff
@@ -45,126 +49,129 @@ class LibtorrentConan(ConanFile):
...
+        tc.variables["i2p"] = self.options.enable_i2p
+        tc.variables["logging"] = self.options.enable_logging
+        tc.variables["mutable-torrents"] = self.options.enable_mutable_torrents
+        tc.variables["build_tests"] = False
+        tc.variables["build_examples"] = False
+        tc.variables["build_tools"] = False
+        tc.variables["python-bindings"] = False
+        tc.variables["python-bindings"] = False
+        if is_msvc(self):
+            tc.variables["static_runtime"] = "MT" in msvc_runtime_flag(self)
```
Comment (SpaceIm) [inline recipes/libtorrent/all/conanfile.py:136]:
```suggestion
            tc.variables["static_runtime"] = is_msvc_static_runtime(self)
```

File: recipes/libtorrent/all/conanfile.py (hunk 3 above)
Comment (toge) [reply recipes/libtorrent/all/conanfile.py]: @SpaceIm 
I combine `_validate_dependency_graph()` to `validate()`.
Thanks.

File: recipes/libtorrent/all/conanfile.py (hunk 4 above)
Comment (toge) [reply recipes/libtorrent/all/conanfile.py]: fixed.

File: recipes/libtorrent/all/conanfile.py (hunk 5 above)
Comment (toge) [reply recipes/libtorrent/all/conanfile.py]: Fixed.

File: recipes/libtorrent/all/conanfile.py (hunk 6 above)
Comment (toge) [reply recipes/libtorrent/all/conanfile.py]: Fixed.
Thanks for your review a lot!

**Review by jwillikers — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**